ui/src/pages/BroadClinicalReview.jsx

```diff
@@ -0,0 +1,215 @@
+import TabPanel from "@mui/lab/TabPanel";
+import { Box, FormControl, InputLabel, Select, MenuItem } from "@mui/material";
+import { DesktopDatePicker } from "@mui/x-date-pickers/DesktopDatePicker";
+import { MobileDatePicker } from "@mui/x-date-pickers/MobileDatePicker";
+import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFns";
+import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
+import CurrentPatient from "../components/CurrentPatient";
+import { useFormik } from "formik";
+import * as yup from "yup";
+import Preview from "../components/Preview";
+import FormFields from "../components/FormFields";
+import broadClinicalReviewFields from "../lib/forms/broadClinicalReview";
+import { apiHost, createEncounter, FhirApi } from "../lib/api";
+
+export default function BroadClinicalReview() {
+  let navigate = useNavigate();
+  let [open, setOpen] = useState(false);
+
+  let [visit, setVisit] = useState();
+  let [message, setMessage] = useState(false);
+  let isMobile = useMediaQuery("(max-width:600px)");
+
+  const [value, setValue] = useState("1");
+
+  const [inputData, setInputData] = useState({});
+  const [preview, setPreview] = useState(false);
+
+  const fieldValues = Object.values(broadClinicalReviewFields).flat();
+  const validationFields = fieldValues
+    .filter((item) => item.validate)
+    .map((item) => ({
+      [item.name]: item.validate,
+    }));
+
+  const validationSchema = yup.object({
+    ...Object.assign({}, ...validationFields),
+  });
+
+
+  const initialValues = Object.assign(
+    {},
+    ...fieldValues.map((item) => ({
+      [item.name]: item.type === "checkbox" ? [] : "",
+    }))
+  );
+
+  const formik = useFormik({
+    initialValues: {
+      ...initialValues,
+    },
+    validationSchema: validationSchema,
+    // submit form
+    onSubmit: (values) => {
+      console.log(values);
+      setPreview(true);
+      setInputData(values);
+    },
+  });
+
+  const prompt=(text)=>{
+
+    setMessage(text);
+    setOpen(true);
+
+    setTimeout(() => {
+      setOpen(false);
+    }, 4000);
+
+    return;
+  }
+
+  let saveBroadClinicalReview = async (values) => {
+    //get current patient
+    let patient = visit.id;
+    if (!patient) {
+      prompt(
+        "No patient visit not been initiated. To start a visit, Select a client from the Client list"
+      );
+      return;
+    }
+
+    //create encounter
+    let encounter = await createEncounter(patient, "BROAD_CLINICAL_REVIEW");
+    // console.log(encounter);
+
+    //save observations
+    //Create and Post Observations
+    let res = await (
+      await FhirApi({
+        url: `/crud/observations`,
+        method: "POST",
+        data: JSON.stringify({
+          patientId: patient,
+          encounterId: encounter.id,
+          observations: values,
+        })
+      })
+    ).data;
+    // console.log(res);
+
+    if (res.status === "success") {
+      prompt("Broad Clinical Review saved successfully");
```

Comment:
After displaying feedback, navigate to patients list where nurse can preview the concluded broad clinical review encounter. Make use of the navigate variable already defined.